src/crypto/x25519.rs

```diff
@@ -0,0 +1,57 @@
+use std::fmt;
+
+use rand::{CryptoRng, Rng};
+use x25519_dalek::{PublicKey, StaticSecret};
+use zeroize::{Zeroize, ZeroizeOnDrop, Zeroizing};
+
+use crate::crypto::{Decryptor, KeyParams};
+use crate::errors::Result;
+use crate::types::{Mpi, PlainSecretParams, PublicParams};
+
+/// Secret key for X25519
+#[derive(Clone, PartialEq, Eq, Zeroize, ZeroizeOnDrop)]
+pub struct SecretKey {
+    pub(crate) secret: [u8; 32],
+}
+
+impl fmt::Debug for SecretKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("X25519SecretKey")
+            .field("secret", &"[..]")
+            .finish()
+    }
+}
+
+impl KeyParams for SecretKey {
+    type KeyParams = ();
+
+    fn key_params(&self) {}
+}
+
+impl Decryptor for SecretKey {
+    // - Ephemeral X25519 public key (32 bytes).
+    // - Recipient public key (32 bytes). [FIXME: use PublicParams and expect PublicParams::X25519?]
+    // - Encrypted and wrapped session key.
+
+    fn decrypt(&self, _mpis: &[Mpi], _fingerprint: &[u8]) -> Result<Vec<u8>> {
+        unimplemented_err!("decrypt for x25519")
```

Comment:
what's the plan here?

Reply:
I plan to do encryption-related functionality right after we're doing with this PR, including x25519 and seipdv2